pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -352,10 +352,23 @@ pub mod pallet {
 							})
 						})?;
 
-					let source_address =
-						Pallet::<T>::try_range(slice_ref, length_source_address, |source_chain| {
-							Ok(source_chain.to_vec())
-						})?;
+					let source_address = Pallet::<T>::try_range(
+						slice_ref,
+						length_source_address,
+						|source_address| {
+							// NOTE: Axelar simply provides the hexadecimal string of an EVM
+							// address as the `sourceAddress` argument.       Solidity does on the
+							// other side recognize the hex-encoding and encode the hex bytes to
+							//       utf-8 bytes.
+							//
+							//       Hence, we are reverting this process here.
+							let source_address =
+								cfg_utils::decode_var_source::<BYTES_ACCOUNT_20>(source_address)
+									.ok_or(Error::<T>::MessageDecodingFailed)?;
```

Comment:
Let's have a specific error type here so that our future selves will have an easier time debugging this if needed ^^